db/version_set.h

```diff
@@ -1160,6 +1161,15 @@ class VersionSet {
       if (min_log_num > num && !cfd->IsDropped()) {
         min_log_num = num;
       }
+      // If mempurge is activated, there may be an immutable memtable
+      // that has data not flushed to any SST file.
+      if (db_options_->experimental_allow_mempurge && !(cfd->IsEmpty()) &&
+          !(cfd->IsDropped())) {
+        num = cfd->imm()->EarliestLogContainingData();
```

Comment:
Its not entirely clear why we need to do this. Shouldn't ```cfd->GetLogNumber()``` be sufficient? The log number in cfd is updated when appending a VersionEdit containing a log number to the manifest, which would only happen on a flush. So the log number in cfd would always be the earliest log with unflushed data I think.

Reply:
You're 100% correct, thanks for pointing that out - I added this code snippet at a time when I didn't think I had to stop the VersionEdit from being appended to the manifest. I will submit an update right now.

Reply:
You're 100% correct, thanks for pointing that out - I added this code snippet at a time when I didn't think I had to stop the VersionEdit from being appended to the manifest. I will submit an update right now.